src/Transports.AspNetCore/AuthorizationHelper.cs

```diff
@@ -0,0 +1,66 @@
+#nullable enable
+
+using System.Security.Claims;
+using System.Security.Principal;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.Server.Transports.AspNetCore;
+
+/// <summary>
+/// Helper methods for performing connection authorization.
+/// </summary>
+public static class AuthorizationHelper
```

Comment:
Why are these classes here and not in Authorization.AspNetCore project?
![изображение](https://user-images.githubusercontent.com/21261007/163689751-b983ed28-af58-4827-b4ba-637f45f76ef6.png)

Reply:
My thoughts:
1. These are authorization rules for the connection, not a validation rule.  They do not scan the schema for metadata or have anything to do with execution of a request.
2. This project is for ASP.Net Core, and these rules fit with the ASP.Net Core authentication and authorization design and interfaces.  Hence there is no need for a separate project.
3. This particular code is also needed by WebSocket code so I put them in their own files.  This is due to WebSocket connections typically do not contain the 'Authorization' HTTP header (since browsers cannot send this header in a WebSocket connection), and so these rules have to be processed only after authentication occurs (during the GQL_CONNECTION_INIT message), and prior to any GraphQL executions.
4. If a user wants to write their own validation code, they can leave the options at their default settings and (optionally) override the proper method in the GraphQLHttpMiddleware to provide their own connection-based authorization. So adding this functionality here has no negative effect if custom authentication or authorization is in use.

Reply:
(In fact, if they override the method, they can use the options in any way they want.)